src/main/java/org/jboss/logging/Log4j2Logger.java

```diff
@@ -52,7 +52,7 @@ protected void doLog(final Level level, final String loggerClassName, final Obje
         if (this.logger.isEnabled(translatedLevel)) {
             try {
                 this.logger.logMessage(loggerClassName, translatedLevel, null,
-                        (parameters == null || parameters.length == 0) ? this.messageFactory.newMessage(message) : this.messageFactory.newMessage(String.valueOf(message), parameters),
+                        (parameters == null || parameters.length == 0) ? this.messageFactory.newMessage(String.valueOf(message)) : this.messageFactory.newMessage(String.valueOf(message), parameters),
```

Comment:
Without the above fix, the following `newMessage()` overloaded method would be chosen:
```
    public Message newMessage(Object message) {
        return new ObjectMessage(message);
    }
```
That explains why the following logging statement
```
log.trace("{}");
```
would end up with the following strange output:
```
{{}}
```
because the object per se would be chosen as the sole parameter for the placeholder (i.e. `{}`):
```
public class ObjectMessage implements Message, StringBuilderFormattable {
... ...
    public Object[] getParameters() {
        return new Object[]{this.obj};
    }
... ...
```